File: tests/07_phases_omp.dg/common/fortran/success_parallel_03.f90
! <testinfo>
! test_generator=config/mercurium-omp
! </testinfo>

PROGRAM P
    IMPLICIT NONE
    INTEGER :: I, N, X
    LOGICAL :: FLAG

    INTEGER, EXTERNAL :: OMP_GET_MAX_THREADS
    X = 0
    N = 1
    FLAG = .FALSE.


    !$OMP PARALLEL IF (.NOT. FLAG .AND. N > 0 ) SHARED(X)
    !$OMP ATOMIC
     X = X + 1
    !$OMP END PARALLEL
    IF (X /= OMP_GET_MAX_THREADS()) STOP 1
END PROGRAM P

File: tests/07_phases_omp.dg/iomp/c/success_parallel_03.c
/*
<testinfo>
test_generator=config/mercurium-iomp
</testinfo>
*/

/*This tests struct shared/private variables*/

#include <assert.h>

int omp_get_num_threads(void);
int omp_get_thread_num(void);

struct A {
    int x;
};

int main(void) {
    struct A a = { -1 };
    #pragma omp parallel num_threads(4) firstprivate(a)
    {
        a.x++;
    }
    assert(a.x == -1);

    #pragma omp parallel num_threads(4)
    {
        a.x++;
    }
    assert(a.x != -1);

}

